libcxx/include/__bit/rotate.h

```diff
@@ -20,24 +20,37 @@
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
+// Writing two full functions for rotl and rotr makes it easier for the compiler
+// to optimize the code. On x86 this function becomes the ROL instruction and
+// the rotr function becomes the ROR instruction.
 template <class _Tp>
-_LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX14 _Tp __rotr(_Tp __t, int __cnt) _NOEXCEPT {
-  static_assert(__libcpp_is_unsigned_integer<_Tp>::value, "__rotr requires an unsigned integer type");
-  const unsigned int __dig = numeric_limits<_Tp>::digits;
```

Comment:
Given that `numeric_limits<_Tp>::digits` (cf. https://en.cppreference.com/w/cpp/types/numeric_limits/digits) already returns an `int` (signed). Wouldn't it be simpler to just do the following (no other changes required if I'm not mistaken)?
```
const int __dig = numeric_limits<_Tp>::digits;
```

Reply:
Seems I didn't submit my comment before. We typically don't use `const` at all. The code was based on the original code. I'll changes it as suggested by Louis.